core/src/main/java/org/mule/DefaultMuleEvent.java

```diff
@@ -605,7 +605,13 @@ public MuleMessage getMessage()
         {
             throw new TransformerException(CoreMessages.objectIsNull("outputType"));
         }
-        return getMuleContext().getTransformationService().getPayload(message, outputType);
+
+        MuleMessage transformedMessage = getMuleContext().getTransformationService().transform(message, outputType);
+        if (ClassUtils.isConsumable(message.getPayload().getClass()))
```

Comment:
I think we did at one point but it got moved out, otherwise message has to have list of consumable types.